sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -650,44 +660,54 @@ def _quick_query_options(self, query_expression,
         if cpk:
             if self.scheme.lower() != 'https':
                 raise ValueError("Customer provided encryption key must be used over HTTPS.")
-            cpk_info = CpkInfo(encryption_key=cpk.key_value, encryption_key_sha256=cpk.key_hash,
-                               encryption_algorithm=cpk.algorithm)
+            cpk_info = CpkInfo(
+                encryption_key=cpk.key_value,
+                encryption_key_sha256=cpk.key_hash,
+                encryption_algorithm=cpk.algorithm
+            )
         options = {
             'query_request': query_request,
             'lease_access_conditions': access_conditions,
             'modified_access_conditions': mod_conditions,
             'cpk_info': cpk_info,
-            'progress_callback': kwargs.pop('progress_callback', None),
             'snapshot': self.snapshot,
             'timeout': kwargs.pop('timeout', None),
             'cls': return_headers_and_deserialized,
-            'client': self._client,
-            'name': self.blob_name,
-            'container': self.container_name}
+        }
         options.update(kwargs)
         return options
 
     @distributed_trace
-    def query(self, query_expression,  # type: str
-              **kwargs):
-        # type: (str, **Any) -> QuickQueryReader
+    def query_blob(self, query_expression, **kwargs):
```

Comment:
I'd prefer query.
The api reside in blob client, then the target should be a blob already.

Reply:
That's true - though I was thinking maybe we should be consistent with `upload_blob` `download_blob` etc

Reply:
I feel query_blob_data is less confusing than query_blob.
Let see see how other people feel?

---
In reply to: [439086909](https://github.com/Azure/azure-sdk-for-python/pull/11991#discussion_r439086909) [](ancestors = 439086909)

Reply:
Talked to folks they don't want to rename it to query_blob, because it will become BlockBlobClient.QueryBlob(), and another class would become BlockBlobClientFactoryQueryBlobFactory

Reply:
Hi Kamil @kasobol-msft , do you have any thought about the api name query_blob?

Reply:
I took a look on the blob client and we seem to have bunch of apis with "blob" and without. I'm not sure what was the reasoning behind having "_blob" suffix in the first place but it seems excessive - as @xiafu-msft we already have reference to "blob client" (I'd ideally drop the suffix from other apis in next major rev).

The Storage team is in favor of keeping consistent name for this feature across the languages.

Reply:
I would really prefer to qualify this name... The name "query" does not actually indicate what the feature does at all, since we can only query the data within specific structured blobs.
Additionally, if other forms of query-like features are introduced in future we could have conflicting terminology.
As @xiafu-msft mentioned - how would you feel about `query_content`?

Reply:
I had an offline chat with @annatisch . Let's leave it as `query_blob`.